src/coreclr/jit/lclmorph.cpp

```diff
@@ -1209,13 +1209,21 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                     assert(elementType == TYP_SIMD12);
                     assert(varDsc->TypeGet() == TYP_SIMD16);
 
+                    // If we are not doing struct promotion and we have zero-initialization,
+                    // then we need to produce a VecCon(0).
+                    if (elementNode->IsIntegralConst(0))
```

Comment:
Just wanted to note that we looked at fixing `CEE_INITOBJ` to do have `TYP_SIMD` be an `ASG` of `ZeroConNode` like it is for primitives

But, it still asserted so there are likely other places also introducing the `ASG = 0` that would also need to be fixed first.